org/postgresql/jdbc2/AbstractJdbc2Array.java

```diff
@@ -934,4 +935,26 @@ public boolean isBinary() {
     public byte[] toBytes() {
         return fieldBytes;
     }
+
+    public int hashCode() {
+        if (fieldBytes != null)
+            return fieldBytes.hashCode();
```

Comment:
His `hashCode` is not consistent with `equals`.
`Arrays#hashCode` should be used to hash the contents of the array.